### DISCUSSION TITLE
Is Vec<Box<T>> redundant?

### DISCUSSION THREAD

**Discussion by Austaras** [swc-project/swc#2107] [Q&A] [answered]
Would Vec<T> be enough?

**Comment by kdy1** [comment, marked as answer]
I think it's not redundant because `Box<Expr>` reduces the size of types containing it.

**Reply by Austaras** [reply]
But is there really a reason to reduce the size of Vec?

**Reply by kdy1** [reply]
I can't remember the exact reason, but I think it was related to `memmove`.
Also, this change requires some profiling.

**Reply by Austaras** [reply]
Oh, maybe it will cost less when move a large struct/enum to a vec if wrapped in a Box.

**Reply by kdy1** [reply]
I remember the reasoning.
`Vec<Box<Expr>>` makes reallocation of vector faster, which is quite common for `Expr` in swc.
